p2p/chain.py

```diff
@@ -218,10 +218,29 @@ def register_peer(self, peer: BasePeer) -> None:
             # TODO: Process headers for consistency.
             try:
                 head_number = await self._process_headers(peer, headers)
+            except HeaderNotFound:
+                # This is comes from when we attempt to compute the difficulty
+                # of a header whos parent is not in our database.
```

Comment:
I'd still like to understand what exactly is going on in #837 before merging a fix, but if you need this fix badly my only suggestion would be to instead doing an explicit check and rewind before calling `_process_headers()`, so we don't have to catch `HeaderNotFound` here, as that might actually hide other bugs in our code. 

Something like
```
    if not await chaindb.coro_header_exists(headers[0].parent_hash):
        start_at = max(0, start_at - eth.MAX_HEADERS_FETCH)
        ...
        continue
    try:
        self._process_headers(...)
        ...
```

Reply:
This is not needed badly and I'm not convinced it is the right solution.  Going to close and wait for this issue to re-surface.